EIP-191 Rework (#337)
LIT-Protocol/docs: docs/sdk/serverless-signing/eip191.md

import FeedbackComponent from "@site/src/pages/feedback.md";

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Signing within a Lit Action using EIP-191

## Overview

Lit Actions offer the ability to use [EIP-191](https://eips.ethereum.org/EIPS/eip-191?ref=blog.spruceid.com) to sign a message instead of a transaction or raw signature. This is done with the Lit Actions method [ethPersonalSignMessageEcdsa](https://actions-docs.litprotocol.com/#ethpersonalsignmessageecdsa). 

> It will prepend `"\x19Ethereum Signed Message:\n"` to the message and then hash and sign it according to the EIP-191 standard.

Reviewer: ```suggestion
It will prepend to your message: `"\x19Ethereum Signed Message:\n"` and the length of the message, then hash and sign it according to the EIP-191 standard.
```

## Prerequisites

> - Basic understanding of [Lit Actions](../serverless-signing/quick-start)

Reviewer: ```suggestion
- Basic understanding of [PKPs](../../../user-wallets/pkps/overview)
- Basic understanding of [Lit Actions](../serverless-signing/quick-start)
```

## Complete Code Example

> The complete code example is available in the [Lit Developer Guides Code Repository](https://github.com/LIT-Protocol/developer-guides-code/tree/master/eip-191-signing). There you can find a Node.js and browser implementation of the code.

Reviewer: ```suggestion
The complete code example is available in the [Lit Developer Guides Code Repository](https://github.com/LIT-Protocol/developer-guides-code/tree/master/eip-191-signing). There you can find a Node.js and browser implementation of this example code.
```

### Example Lit Action

> The following Lit Action uses `ethPersonalSignMessageEcdsa` to append `"\x19Ethereum Signed Message:\n"` to the given message and then hash and sign it according to the EIP-191 standard. The `sigShare` variable is set to a boolean value of `true` if the signature was successfully generated, and `false` otherwise. The Lit Action will also return the complete signature as an object under the `sigName` key.

Reviewer: ```suggestion
The return value of `ethPersonalSignMessageEcdsa` (the `sigShare` variable in this example) is set to a boolean value of `true` if the signature was successfully generated, and `false` otherwise. The Lit Action will also return the complete signature as an object under the `sigName` key.
```
